cloudinit/sources/DataSourceEc2.py

```diff
@@ -97,10 +104,23 @@ class DataSourceEc2(sources.DataSource):
         }
     }
 
+    default_update_events = {
+        EventScope.NETWORK: {
```

Comment:
Seeing it here, I'm not sure this combination makes sense. This essentially says "use the cached network config unless we get a hotplug event", but that means that if a network interface is added while the machine is powered off, we won't see it on a subsequent boot.

Not something to block this PR, but I think we need to consider adding `EventType.BOOT` here as well.

Reply:
Thanks for pointing this out and create https://bootstack.canonical.com/cases/00378972. This will be handled separately.